storage/lrucache/capacity/capacityLRUCache.go

```diff
@@ -0,0 +1,252 @@
+package capacity
+
+import (
+	"container/list"
+	"fmt"
+	"sync"
+
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+	"github.com/ElrondNetwork/elrond-go/storage"
+)
+
+var log = logger.GetOrCreate("storage/lrucache/capacity")
+
+// CapacityLRU implements a non thread safe LRU Cache with a max capacity size
+type CapacityLRU struct {
+	lock                   sync.Mutex
+	size                   int
+	maxCapacityInBytes     int64
+	currentCapacityInBytes int64
+	evictList              *list.List
+	items                  map[interface{}]*list.Element
+}
+
+// entry is used to hold a value in the evictList
+type entry struct {
+	key   interface{}
+	value interface{}
+	size  int64
+}
+
+// NewCapacityLRU constructs an CapacityLRU of the given size with a byte size capacity
+func NewCapacityLRU(size int, byteCapacity int64) (*CapacityLRU, error) {
+	if size < 1 {
+		return nil, storage.ErrCacheSizeInvalid
+	}
+	if byteCapacity < 1 {
+		return nil, storage.ErrCacheCapacityInvalid
+	}
+	c := &CapacityLRU{
+		size:               size,
+		maxCapacityInBytes: byteCapacity,
+		evictList:          list.New(),
+		items:              make(map[interface{}]*list.Element),
+	}
+	return c, nil
+}
+
+// Purge is used to completely clear the cache.
+func (c *CapacityLRU) Purge() {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	c.items = make(map[interface{}]*list.Element)
+	c.evictList.Init()
+	c.currentCapacityInBytes = 0
+}
+
+// AddSized adds a value to the cache.  Returns true if an eviction occurred.
+func (c *CapacityLRU) AddSized(key, value interface{}, sizeInBytes int64) bool {
+	if sizeInBytes < 0 {
+		log.Error("size LRU cache add error",
+			"key", fmt.Sprintf("%v", key),
+			"value", fmt.Sprintf("%v", value),
+			"error", "size in bytes is negative",
+		)
+
+		return false
+	}
+
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	// Check for existing item
+	if ent, ok := c.items[key]; ok {
+		c.update(key, value, sizeInBytes, ent)
+	} else {
+		c.addNew(key, value, sizeInBytes)
+	}
+
+	return c.evictIfNeeded()
+}
+
+func (c *CapacityLRU) addNew(key interface{}, value interface{}, sizeInBytes int64) {
+	ent := &entry{key, value, sizeInBytes}
```

Comment:
please name the fields. (`key: key` for example)

Reply:
done